### PR TITLE
Desktop: Add network proxy ignore setting

### PR DESCRIPTION
**Pull request by rihardsgravis**
# Description

Add system wide network proxy setting bypass setting.

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

Tested manually on macOS

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

File: src/desktop/main.js [hunk 1]
```diff
@@ -72,9 +72,18 @@ try {
     if (windowStateData) {
         windowState = windowStateData;
     }
-    settings = JSON.parse(data);
+    settings = JSON.parse(data) || {};
 } catch (error) {}
 
+/**
+ * Temporary disable proxy if not overrided by settings
```
Comment (rajivshah3) [inline src/desktop/main.js]: `Temporary` -> `Temporarily`
`overrided` -> `overridden`

File: src/shared/locales/en/translation.json [hunk 2]
```diff
@@ -737,6 +737,10 @@
          "setupIncomplete": "Wallet setup incomplete",
          "completeSetup": "Complete setup"
     },
+    "proxy": {
+        "proxy": "Use system proxy",
+        "proxyExplanation": "Detect and use system wide network proxy settings. Restart the wallet after setting change."
```
Comment (rajivshah3) [inline src/shared/locales/en/translation.json]: Maybe this should be changed to `Restart the wallet after changing this setting`?

**Comment by rajivshah3** [comment]
Would this PR solve https://github.com/iotaledger/trinity-wallet/issues/503?

**Review by cvarley100 — APPROVED** [review]

**Comment by rihardsgravis** [comment]
> Would this PR solve #503?

No. This change allows to ignore system wide proxy settings. That issue requests for manual proxy input.

**Comment by rajivshah3** [comment]
> No. This change allows to ignore system wide proxy settings. The issue requests for manual proxy input.

Ah, I see. I was under the impression that this would allow it to detect the system's proxy settings. I might be able to try that in a future PR.

**Review by rajivshah3 — APPROVED** [review]
Looks good! 👍

**Comment by rihardsgravis** [comment]
> > No. This change allows to ignore system wide proxy settings. The issue requests for manual proxy input.
> 
> Ah, I see. I was under the impression that this would allow it to detect the system's proxy settings. I might be able to try that in a future PR.

Electron detects system proxy settings by default. A fix for #503 could be manual proxy input at Trinity Advanced settings.

**Pull request merged**